File: leetcode/001_TwoSum/TwoSum.cpp
#include <unordered_map>

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> index_map;

        for (int i = 0; i < nums.size(); ++i) {
            int difference = target - nums[i];

            if (index_map.find(difference) != index_map.end()) {
                return {index_map[difference], i};
            }

            index_map[nums[i]] = i;
        }
        return {};
    }
};

File: leetcode/001_TwoSum/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_dict = {}
        
        for i, num in enumerate(nums):
            difference = target - num
            
            if difference in index_dict:
                return [index_dict[difference], i]
            
            index_dict[num] = i